components/Ripple/src/MDCStatefulRippleView.m

```diff
@@ -54,37 +55,45 @@ - (instancetype)initWithCoder:(NSCoder *)coder {
 }
 
 - (void)commonMDCStatefulRippleViewInit {
-  if (_rippleColors == nil) {
+  if (_rippleDefaultColors == nil) {
     _rippleColors = [NSMutableDictionary dictionary];
+    _rippleDefaultColors = [NSMutableDictionary dictionary];
     UIColor *selectionColor = RippleSelectedColor();
-    _rippleColors[@(MDCRippleStateNormal)] = [UIColor colorWithWhite:0 alpha:kDefaultRippleAlpha];
-    _rippleColors[@(MDCRippleStateHighlighted)] = [UIColor colorWithWhite:0
-                                                                    alpha:kDefaultRippleAlpha];
-    _rippleColors[@(MDCRippleStateSelected)] =
+
+    _rippleDefaultColors[@(MDCRippleStateNormal)] = [UIColor colorWithWhite:0
+                                                                      alpha:kDefaultRippleAlpha];
+    _rippleDefaultColors[@(MDCRippleStateHighlighted)] =
+        [UIColor colorWithWhite:0 alpha:kDefaultRippleAlpha];
+    _rippleDefaultColors[@(MDCRippleStateSelected)] =
         [selectionColor colorWithAlphaComponent:kDefaultRippleSelectedAlpha];
-    _rippleColors[@(MDCRippleStateSelected | MDCRippleStateHighlighted)] =
+    _rippleDefaultColors[@(MDCRippleStateSelected | MDCRippleStateHighlighted)] =
         [selectionColor colorWithAlphaComponent:kDefaultRippleAlpha];
-    _rippleColors[@(MDCRippleStateDragged)] = [UIColor colorWithWhite:0
-                                                                alpha:kDefaultRippleDraggedAlpha];
-    _rippleColors[@(MDCRippleStateDragged | MDCRippleStateHighlighted)] =
+    _rippleDefaultColors[@(MDCRippleStateDragged)] =
+        [UIColor colorWithWhite:0 alpha:kDefaultRippleDraggedAlpha];
+    _rippleDefaultColors[@(MDCRippleStateDragged | MDCRippleStateHighlighted)] =
         [UIColor colorWithWhite:0 alpha:kDefaultRippleDraggedAlpha];
-    _rippleColors[@(MDCRippleStateSelected | MDCRippleStateDragged)] =
+    _rippleDefaultColors[@(MDCRippleStateSelected | MDCRippleStateDragged)] =
         [selectionColor colorWithAlphaComponent:kDefaultRippleDraggedAlpha];
   }
 }
 
 - (UIColor *)rippleColorForState:(MDCRippleState)state {
-  UIColor *rippleColor = _rippleColors[@(state)];
+  UIColor *rippleColor = _rippleColors[@(state)] ?: _rippleDefaultColors[@(state)];
```

Comment:
This logic is very complex and behaves very differently from UIControl (the only stateful API I'm really familiar with). Could you please add comments/docs to the method to help outline the updated logic?